crates/store/re_entity_db/src/entity_db.rs

```diff
@@ -629,6 +663,72 @@ impl EntityDb {
         self.set_store_info = Some(store_info);
     }
 
+    /// Submit an Arrow message to the background ingestion worker (native only).
+    ///
+    /// On native: Lazily creates worker on first call, then queues message for processing
+    /// On Wasm: This method doesn't exist (compile error if called)
+    ///
+    /// The caller should periodically call [`Self::poll_worker_output`] to retrieve
+    /// processed chunks and add them to the store.
+    #[cfg(not(target_arch = "wasm32"))]
+    pub fn submit_arrow_msg(
+        &mut self,
+        arrow_msg: re_log_types::ArrowMsg,
+        channel_source: Arc<re_smart_channel::SmartChannelSource>,
+        msg_will_add_new_store: bool,
+    ) {
+        re_tracing::profile_function!();
+
+        // Lazy-init worker on first use
+        if self.ingestion_worker.is_none() {
+            re_log::debug!("Creating ingestion worker for store {:?}", self.store_id);
+            self.ingestion_worker = Some(crate::ingestion_worker::IngestionWorker::new());
+        }
+
+        if let Some(worker) = &self.ingestion_worker {
+            worker.submit_arrow_msg_blocking(
+                self.store_id.clone(),
+                arrow_msg,
+                channel_source,
+                msg_will_add_new_store,
+            );
+        }
+    }
```

Comment:
Good opportunity to use `Option::get_or_insert_with` here. It better expresses the intent and doesn't hide the important code behind a scary `if`